File: src/alignment/alignment.go
// alignment.go
//
// Copyright (c) 2026 PDFjet Software
// Licensed under the MIT License. See LICENSE file in the project root.

// Package alignment defines the horizontal and vertical alignments.
package alignment

// Alignment specifies the horizontal or vertical alignment. It is an int32,
// so that the three of them a Cell holds cost 4 bytes each and not 8.
type Alignment int32

// Left, Right, Center and Justify align horizontally; Top, Center and Bottom
// align vertically.
const (
	Left Alignment = iota
	Right
	Center
	Justify
	Top
	Bottom
)

File: Sources/PDFjet/Alignment.swift
/**
 * Alignment.swift
 *
 * Copyright (c) 2026 PDFjet Software
 * Licensed under the MIT License. See LICENSE file in the project root.
 */

/**
 * Used to specify the horizontal and vertical alignment, for example of the
 * text in a Cell, TextBlock, Paragraph or TextColumn.
 */
/// The values are the same as the ordinals of the other three ports, so that
/// a Cell can keep three alignments in the bits of one integer.
public enum Alignment: UInt32 {
    /// Aligns to the left.
    case LEFT = 0
    /// Aligns to the right.
    case RIGHT = 1
    /// Centers horizontally or vertically.
    case CENTER = 2
    /// Justifies the text.
    case JUSTIFY = 3
    /// Aligns to the top.
    case TOP = 4
    /// Aligns to the bottom.
    case BOTTOM = 5
}
